diff -Npaur Configuration-old/api/module.php Configuration/api/module.php
--- Configuration-old/api/module.php	2017-09-22 18:14:50.000000000 +0100
+++ Configuration/api/module.php	2017-09-22 18:42:51.000000000 +0100
@@ -48,6 +48,15 @@ class Configuration extends SystemModule
             case 'disableLandingPage':
                 $this->disableLandingPage();
                 break;
+
+	    case 'changeRegion':
+                $this->changeRegion();
+                break;
+
+            case 'getCurrentRegion':
+                $this->getCurrentRegion();
+                break;
+
         }
     }
 
@@ -133,4 +142,17 @@ class Configuration extends SystemModule
 
         $this->response = array("success" => false);
     }
+
+    private function getCurrentRegion()
+    {
+        $currentRegion = exec('iw reg get|head -n 2|tail -n 1|cut -b 9-10');
+        $this->response = array("currentRegion" => $currentRegion);
+    }
+
+    private function changeRegion()
+    {
+        $region = escapeshellarg($this->request->Region);
+        exec("iw reg set {$region}");
+        $this->response = array("success" => true);
+    }
 }
diff -Npaur Configuration-old/js/module.js Configuration/js/module.js
--- Configuration-old/js/module.js	2017-09-22 18:14:50.000000000 +0100
+++ Configuration/js/module.js	2017-09-22 18:42:51.000000000 +0100
@@ -7,6 +7,8 @@ registerController("ConfigurationGeneral
 	$scope.newPasswordRepeat = "";
 	$scope.showPasswordSuccess = false;
 	$scope.showPasswordError = false;
+	$scope.customRegion="";
+	$scope.currentRegion="";
 
 	$scope.timeZones = [
 		{ value: 'GMT+12', description: "(GMT-12:00) Eniwetok, Kwajalein" },
@@ -36,6 +38,15 @@ registerController("ConfigurationGeneral
 		{ value: 'GMT-12', description: "(GMT+12) Auckland, Wellington, Fiji, Kamchatka" }
 	];
 
+        $scope.getCurrentRegion = (function() {
+                $api.request({
+                        module: "Configuration",
+                       action: "getCurrentRegion"
+                }, function(response) {
+                        $scope.currentRegion = response.currentRegion;
+               });
+        });
+
 
 	$scope.getCurrentTimeZone = (function() {
 		$api.request({
@@ -137,6 +148,25 @@ registerController("ConfigurationGeneral
 		});
 	});
 
+        $scope.changeRegion = (function() {
+		var tmpRegion;
+		if ($scope.customRegion.trim() !== "") {
+                        tmpRegion = $scope.customRegion;
+                }
+                $api.request({
+                        module: "Configuration",
+                        action: "changeRegion",
+                        Region: tmpRegion,
+
+                }, function(response) {
+                        if (response.success !== undefined) {
+                                $scope.getCurrentRegion();
+				$scope.customRegion="00";
+                        }
+                });
+        });
+
+	$scope.getCurrentRegion();
 	$scope.getCurrentTimeZone();
 }]);
 
@@ -195,4 +225,4 @@ registerController('ConfigurationLanding
 	});
 
 	$scope.getLandingPageStatus();
-}]);
\ No newline at end of file
+}]);
diff -Npaur Configuration-old/module.html Configuration/module.html
--- Configuration-old/module.html	2017-09-22 18:14:50.000000000 +0100
+++ Configuration/module.html	2017-09-22 18:42:51.000000000 +0100
@@ -43,7 +43,23 @@
                         </div>
                     </div>
                 </form>
-
+                <br/>
+		<form class="form-horizontal">
+			<div class="form-group">
+			<label class="col-sm-2 control-label">Region:</label>
+			<div class="col-sm-3">
+			<input type="text" class="form-control" ng-model="currentRegion" disabled>
+			</div></div>
+			<div class="form-group">
+			<label for="Region" class="col-sm-2 control-label">Region Code</label>
+			<div class="col-sm-5">
+			<input type="text" class="form-control" placeholder="00" ng-model="customRegion">
+			</div></div>
+			<div class="form-group">
+			<div class="col-sm-offset-2 col-sm-10">
+			<button type="submit" class="btn btn-default" ng-click="changeRegion()">Change Region</button>
+			</div></div>
+		</form>
                 <br/>
                 <br/>
 
@@ -101,4 +117,4 @@
             </div>
         </div>
     </div>
-</div>
\ No newline at end of file
+</div>
